Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputField.swift

```diff
@@ -0,0 +1,92 @@
+/// For the Credit Card feature, this enum holds four cases that describe different parts of a credit card.
+enum CreditCardInputType {
+    case name, number, expiration, securityCode
+}
+
+struct CreditCardInputField: View {
+    struct Colors {
+        let errorColor: Color
+        let titleColor: Color
+        let textFieldColor: Color
+    }
+
+    @Binding var text: String
+    @ObservedObject var viewModel: CreditCardInputViewModel
+    let creditCardInputType: CreditCardInputType
+    var colors: Colors
+
+    init(creditCardInputType: CreditCardInputType,
+         text: Binding<String>,
+         colors: Colors) {
+        self.creditCardInputType = creditCardInputType
+        self.viewModel = CreditCardInputViewModel(fieldType: creditCardInputType)
+        self._text = text
+        self.colors = colors
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            getInputFieldViewWith(inputType: creditCardInputType)
+        }.padding(.leading, 16)
+    }
+
+    @ViewBuilder private func getInputFieldViewWith(inputType: CreditCardInputType) -> some View {
+        Text(viewModel.fieldHeadline)
+            .font(.subheadline)
+            .foregroundColor(colors.titleColor)
+        TextField("", text: $text)
+            .font(.body)
+            .foregroundColor(colors.textFieldColor)
+            .keyboardType(viewModel.keyboardType)
+            .onChange(of: text) { [oldValue = text] newValue in
+                switch inputType {
+                case .name:
+                    viewModel.updateNameValidity(userInputtedText: text)
+                case .number:
+                    viewModel.updateCardNumberValidity(userInputtedText: text)
+                case .expiration:
+                    var numbersCount = 0
+                    text.forEach { character in
+                        character.isNumber ? numbersCount += 1 : nil
+                    }
+
+                    guard !(text.count > viewModel.formattedTextLimit) || !(numbersCount > 5) else {
+                        text = oldValue
+                        return
+                    }
+
+                    guard text.count % 4 == 0 else {
+                        text = text.removingOccurrences(of: " / ")
+                        return
+                    }
+
+                    viewModel.updateExpirationValidity(userInputtedText: text)
+
+                    guard let formattedText = viewModel.separate(inputType: inputType, for: text) else { return }
+                    text = formattedText
+                case .securityCode:
+                    viewModel.updateSecurityCodeValidity(userInputtedText: text)
+                }
+            }
+        if !viewModel.isValid {
+            errorViewWith(errorString: viewModel.errorString)
+        }
+    }
+
+    @ViewBuilder private func errorViewWith(errorString: String) -> some View {
+        HStack(spacing: 0) {
+            Image(ImageIdentifiers.errorAutofill)
```

Comment:
The image name is read in VoiceOver. So either it needs an a11y label or it should be excluded. If it gets an a11y label, both the image and label should be grouped together so they are read together instead of as separate elements.

Reply:
Similar to how we've been handling it in other areas, I chose to exclude it. Thanks for pointing that out!